src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -788,15 +795,16 @@ private void initializeVerifier() {
      */
     private byte[] getBytes(ZipEntry ze) throws IOException {
         try (InputStream is = super.getInputStream(ze)) {
-            int len = (int)ze.getSize();
+            long len = ze.getSize();
+            if (len > MAX_BUFFER_SIZE) {
+                throw new OutOfMemoryError("Required array size too large");
+            }
```

Comment:
I would just add a new `long zeSize` to read and check `ze.getSize()`, and then (int) cast it into `len`, as before.  Then I think no changes would be needed past L802, `int bytesRead;`

Reply:
Done. Changed it based on your input.